build.sbt

```diff
@@ -53,6 +54,8 @@ lazy val daffodil = project
     udf
   )
   .settings(commonSettings, nopublish, ratSettings, unidocSettings, genCExamplesSettings)
+  .settings(javadocSettings: _*) // for genjavadoc to create javadoc from scala code
```

Comment:
@stevedlawrence this line is one of the permutations of how to make genjavadoc work from the github readme for that tool. I also tried using `.config(Javadoc)` from their instructions also, with no effect. 

This doesn't cause an error, but neither does it work to cause the genjavadoc and unidoc to work together.

Reply:
My guess is we need something similar to `daffodil-japi/build.sbt` copied into `daffodil-runtime1` to enable the GenJavadoc plugin for sbt-unidoc to use.

Reply:
Please take another look. 

This runs, but errors in genjavadoc becuase the filters do not apply to runtime1 files until after it fails with a zillion javadoc errors trying to convert scaladoc to javadoc for files we do not care about. We need to exclude various directories before genjavadoc converts them, but currently genjavadoc by way of unidoc, is getting handed whole sub-projects like "runtime1" and "japi" to convert.

Reply:
I've just pushed a change that I think fixes the CI, looks like the CI is passing so far.

This fixes it by disabling publishing of -doc jar artifacts, including genjavadoc artifacts. We don't really need these and people probably shouldn't reference them anyways. They should use the public API published on our website. The reason this works:

There is no way to filter out what GenJavadoc generates. When SBT compiles a scala file with the genjavadoc compiler plugin enabled it is going to generate a java file for it, no matter what. For complex scala files like in runtime1, this can sometimes generate invalid java files that cause scala/javadoc to fail. And when scaladoc/javadoc is run it's going to cause a failure.

However, we use unidoc which runs on a subset of files (thanks to the new filtering) so these invalid files should be ignored. However, when publishing jars (which a number a tasks can trigger like publishLocal, daffodil-cli/stage, etc.), it also publishes -doc artifacts for each project. These doc artifacts do not use the same filtering as unidoc, so they end up using these invalid files and cause the failure.

Removing the PublishJavaDocPlugin is probably sufficient to cause it to ignore these invalid java files when publishing -doc artifacts, but really we don't need to be publishing any docs.